Fix #2419: Updated navigation drawer tests in HelperFragmentTest

srikanthkb opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes #2419 
Fixes #2257

Updated a new method to be used for navigation drawer opening in the `HelpFragmentTest` and `OptionsFragmentTest`, to support RoboElectric tests.
Screenshot of the tests with the changes is attached below:

![image](https://user-images.githubusercontent.com/36214903/103889748-ee09a900-510c-11eb-9075-e64c59c227a1.png)

![image](https://user-images.githubusercontent.com/36214903/104686338-038d5d00-5723-11eb-9057-362bdfa898b2.png)

Removed unnecessary `FirebaseApp.initialization()` from the test files to support Roboelectric and Esspresso tests completion.
## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

srikanthkb committed:
Updated the tests in HelperFragmentTest

Created a private method to perform opening of navigation drawer
in HelperFragmentTest to support RoboElectric tests

srikanthkb commented on app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt:
```diff
@@ -171,12 +171,35 @@ class HelpFragmentTest {
+      // NOTE TO DEVELOPERS: if this ever flakes, we can probably put this in a loop with fake time
+      // adjustments to simulate the render loop.
+      drawerLayout.computeScroll()
+    }
```
@anandwana001 
I have not included the `testCoroutineDispatchers.runCurrent()` here.
Is it required ? if yes, we have to inject the `testCoroutineDispatchers` into HelperFragmentTest

anandwana001 replied:
Let's add it, this is something which stabilizes the test. 
Could you please also add a screenshot for robolectric test result?

anandwana001 commented:
Let's find if any other test fil has a similar issue.

srikanthkb committed:
Update HelperFragmentTest

* Remove Firebase initialization (caused error while running
  robo-electric test
* Use testCoroutineDispatchers
* Updated isClosed() test to run explicitly on espresso only

srikanthkb committed:
Remove FirebaseApp.initializeApp()

Remove the FirebaseApp.initializeApp() from setUp() to support
successful completion of robo-electric tests on local PC

srikanthkb committed:
Merge remote-tracking branch 'upstream/develop' into update-helper-fragment-tests

rt4914 commented:
@Srikanth-Kb Make sure that the `Assignees` section is up-to-date. 
https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

srikanthkb commented:
> 
> 
> @Srikanth-Kb Make sure that the `Assignees` section is up-to-date.
> https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

Noted. Thanks.

rt4914 reviewed (approved):
LGTM

anandwana001 reviewed (changes requested):
`testOptionFragment_clickNavigationDrawerHamburger_navigationDrawerIsOpenedSuccessfully`

need to fix this test case as well.

srikanthkb commented:
> `testOptionFragment_clickNavigationDrawerHamburger_navigationDrawerIsOpenedSuccessfully`
> 
> need to fix this test case as well.

Got it.. I will check for other occurrences and update. Thank you @anandwana001

srikanthkb committed:
Merge remote-tracking branch 'upstream/develop'

anandwana001 commented:
@Srikanth-Kb any update on this?

srikanthkb committed:
Updated OptionsFragmentTest with similar approach

srikanthkb committed:
Merge remote branch upstream/develop

srikanthkb commented:
> @Srikanth-Kb any update on this?

Updated the PR with changes.

srikanthkb committed:
Created helper class for scrolling actions

* Created a new helper class including methods for all of required
  RecyclerView actions for the codebase
* Added required gradle dependencies in the testing module

srikanthkb committed:
Merge remote-tracking branch

srikanthkb committed:
Revert "Created helper class for scrolling actions"

This reverts commit a915148574cda6d72312700605cfd9dea8473845.

srikanthkb committed:
Merge remote-tracking branch

srikanthkb commented:
@anandwana001 requesting your review.

anandwana001 reviewed (approved):
LGTM
